### PR TITLE
feat(em) add export csv option broken down by precinct and voting method

### PR DESCRIPTION
**Pull request by carolinemodic**
## Overview

https://github.com/votingworks/vxsuite/issues/2317
Adds an export results option that saves a CSV of results broken down by precinct and voting method

## Demo Video or Screenshot

https://user-images.githubusercontent.com/14897017/184276313-c3a49578-acc6-41cd-9591-dc2a32483c21.mov

Example Output: 
[votingworks-test-results_sample-county_example-primary-election_2022-08-11_20-22-26.csv](https://github.com/votingworks/vxsuite/files/9313554/votingworks-test-results_sample-county_example-primary-election_2022-08-11_20-22-26.csv)

## Testing Plan 
Manually exported files and sanity checked, one simple snapshot-esque test
Needs more manually testing tomorrow 

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [ ] I have added JSDoc comments to any newly introduced exports

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: frontends/election-manager/src/screens/reports_screen.tsx [hunk 1]
```diff
@@ -188,10 +189,17 @@ export function ReportsScreen(): JSX.Element {
...
-            {hasConverter && castVoteRecordFiles.wereAdded && (
-              <Button onPress={() => setIsExportResultsModalOpen(true)}>
-                Save Results File
-              </Button>
+            {castVoteRecordFiles.wereAdded && (
+              <React.Fragment>
+                {hasConverter && (
+                  <React.Fragment>
+                    <Button onPress={() => setIsExportResultsModalOpen(true)}>
+                      Save SEMS Results
```
Comment (benadida) [inline frontends/election-manager/src/screens/reports_screen.tsx:197]: Can we make the text here dependent on which converter we're using? SEMS is specific to Mississippi. We have a similar conditional for the converter button.

File: frontends/election-manager/src/utils/generate_results_csv.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    `"${precinct.name}"`,
+    precinct.id,
+    votingMethod,
+    votes?.toString() ?? '0',
+    `"${contest.title}"`,
+    contest.id,
+    `"${selection}"`,
+    selectionId ?? '',
+  ].join(',');
+}
```
Comment (benadida) [inline frontends/election-manager/src/utils/generate_results_csv.ts:30]: feels like we should be using an existing `csv` library so we don't have to do our own escaping, and we handle various character encoding edge cases automatically.

**Review by benadida — APPROVED** [review]
two suggested changes:
- use CSV exporter that does all escaping for us
- don't hard-wire SEMS as only option, make it conditional on which exporter we're using.

**Comment by mattroe** [comment]
I had @ginvdr sanity check this; two small requests @carolinemodic:
- Change `Ballots` to `Votes`. 
- Order columns left to right as: `Contest`, `Candidate`, `Precinct`, `Voting Method`, `Votes`

**Review by mattroe — COMMENTED** [review]
I had @ginvdr sanity check this; two small requests @carolinemodic:

Change Ballots to Votes.
Order columns left to right as: Contest, Candidate, Precinct, Voting Method, Votes

**Pull request merged**